File: leetcode_problems/python/lc1598_crawler_log_folder.py
class Solution:
    def minOperations(self, logs: list[str]) -> int:
        steps = 0
        for log in logs:
            if log == "../":
                if steps > 0:
                    steps -= 1
            elif log == "./":
                pass
            else:
                steps += 1

        return steps


if __name__ == "__main__":
    logs = ["d1/", "d2/", "../", "d21/", "./"]
    print(Solution().minOperations(logs))

File: leetcode_problems/cpp/lc1598_crawler_log_folder.cpp
#include <vector>
#include <string>
#include <stack>
using std::stack;
using std::string;
using std::vector;

class Solution
{
public:
    int minOperations(vector<string> &logs)
    {
        stack<string> st;
        for (const string &log : logs)
        {
            if (log == "../")
            {
                if (!st.empty())
                {
                    st.pop();
                }
            }
            else if (log == "./")
            {
                continue;
            }
            else
            {
                st.push(log);
            }
        }
        return st.size();
    }
};